modules/core/monitor/storekit/elasticsearch/index/cleaner/clean.go

```diff
@@ -136,3 +146,47 @@ func (p *provider) deleteIndex(indices []string) error {
 	p.Log.Infof("clean indices %d, %v", len(indices), indices)
 	return nil
 }
+
+func (p *provider) deleteByQuery() {
+	ctx := context.Background()
+	p.loader.WaitAndGetIndices(ctx)
+	p.Log.Infof("Running ES docs clean...")
+	defer p.Log.Infof("Docs cleaned...")
+
+	indexGroup := p.loader.AllIndices()
+	if indexGroup == nil {
+		return
+	}
+	indices := p.getIndicesList(ctx, indexGroup)
+
+	for _, index := range indices {
+		ttl := elastic.NewRangeQuery("@timestamp").From(0).To(time.Now().AddDate(0, 0, -p.Cfg.DiskClean.TTL.MaxStoreTime).UnixNano() / 1e6)
+		resp, err := p.loader.Client().DeleteByQuery(index).WaitForCompletion(false).Conflicts("proceed").Query(ttl).Do(ctx)
+		if err != nil {
+			p.forceMerge(ctx, index)
+			p.Log.Errorf("delete failed. indices: %s, err: %v", index, err.Error())
+			continue
+		}
+		if resp.Failures != nil && len(resp.Failures) > 0 {
+			p.forceMerge(ctx, index)
+			p.Log.Errorf("delete failed. err: %v", resp.Failures)
+			continue
+		}
+
+		p.Log.Infof("Clean doc by ttl, indices count: %d, indices %v, docs count: %v ", len(index), index, resp.Total)
+		p.forceMerge(ctx, index)
```

Comment:
1. the `forceMerge` got called too early, becasue the deleting-query may still running